zebra-chain/src/block.rs

```diff
@@ -104,22 +115,34 @@ pub struct BlockHeader {
     nonce: [u8; 32],
 
     /// The Equihash solution.
-    // The solution size when serialized should be in bytes ('always
-    // 1344').  I first tried this as a [u8; 1344] but until const
-    // generics land we'd have to implement all our common traits
-    // manually, like in pzec.
-    solution: Vec<u8>,
+    solution: EquihashSolution,
 }
 
 impl ZcashSerialize for BlockHeader {
-    fn zcash_serialize<W: io::Write>(&self, _writer: W) -> Result<(), SerializationError> {
-        unimplemented!();
+    fn zcash_serialize<W: io::Write>(&self, mut writer: W) -> Result<(), SerializationError> {
```

Comment:
Where do we write `nVersion`? Per https://zips.z.cash/protocol/protocol.pdf#blockheader it looks like there should be a leading `int32` with value `4`. Since

> The current and only defined block version number for Zcash is 4.

I don't think we need to store it internally, but it does seem like we would need to write it.